content_scripts/ui_component.coffee

```diff
@@ -39,13 +60,21 @@ class UIComponent
     @iframeElement.focus()
 
   show: (message) ->
+    unless @loaded
+      @queueAction "show", arguments
+      return
+
     @postMessage message if message?
     @iframeElement.classList.remove "vimiumUIComponentHidden"
-    @iframeElement.classList.add "vimiumUIComponentShowing"
+    @iframeElement.classList.add "vimiumUIComponentVisible"
     @showing = true
 
-  hide: (focusWindow = true)->
-    @iframeElement.classList.remove "vimiumUIComponentShowing"
+  hide: (focusWindow = true, forceImmediate = false) ->
+    unless @loaded or forceImmediate
+      @queueAction "hide", arguments
+      return
+
+    @iframeElement.classList.remove "vimiumUIComponentVisible"
     @iframeElement.classList.add "vimiumUIComponentHidden"
     window.focus() if focusWindow
```

Comment:
That works: a3909bafd3a38d2d4cd780c757a136e2e0f20dee.
Your work now copied [here](https://github.com/smblott-github/vimium/tree/mrmr1993-hud-to-frame-uicomponent).